### DISCUSSION TITLE
Why does FOOOCUS always put female sex organs on men?

### DISCUSSION THREAD

**Discussion by JRprof53** [lllyasviel/Fooocus#2806] [Q&A] [answered]
I absolutely love FOOOCUS. I thank the creators. Creating accurate, realistic photo quality images of females, no problem.  I'm an artist, as well as a content creator. Everytime I prompt FOOOCUS to generate female nude images, all is correct and accurate. However, when I ask for nude male images, FOOOCUS returns accurate images of nude men, except, well, with the sex organs of a female. Every time. I have tried tweaking the prompts, adding negative prompts, and no matter what, it always generates a male with female sex organs. I've tried unchecking the default Styles, unchecking various Models, unchecked Random, and every other option I'm aware of. I've also added strength to specific words, and it still won't work. And this same thing occurs on 2 separate computers. I would really appreciate help from anybody. Thanks in advance.

**Comment by ILikeToasters** [comment, marked as answer]
Because the models where not trained on male genitalia. Has nothing to do with Fooocus itself. You can try out using LoRAs for that or I do know the latest JuggernautX can do them if you ask for it specifically. As they note "Some areas require multiple attempts to achieve a good result. Especially notable are the "Text","Male Genitalia" and "Hand Poses" areas. All are possible with the model but are still in a "development phase.""
https://civitai.com/models/133005/juggernaut-xl
Prompting guide for JuggernautX: https://storage.googleapis.com/run-diffusion-public-assets/Prompting_Juggernaut_X.pdf

**Reply by JRprof53** [reply]
Thanks so much for taking the time to answer. Tried all you suggested, and still, female genitalia on every male body. Just don't know how to fix this. Thanks again. I appreciate it.

**Reply by KarlJovanick** [reply]
Don't forget that we live in a predominantly heterosexual society that pretends to ignore the fact that 10% of the population has a different sexual orientation. Although there is certainly interest in an alternative to this dominance, creators prefer to ignore it for fear of censorship initiated by the European DSA and the American conservative right. It is therefore not surprising that most NSFW models are trained on female bodies. Indeed, "civitai" accepts male NSFW models on its site, but they are rare compared to other Lora or models. I hesitate to give the names of the models; I would risk being banned from GitHub, which seems to be subject to a form of censorship that prioritizes a certain conservative morality at the expense of freedom.

**Comment by JRprof53** [comment]
Hi there. Sorry for the delay in thanking you again for getting back to me
a second time.
Must be a black hole I entered when I moved back to NJ from Fla.
Not sure why I am having so much difficulty in being even close to
achieving my goal with FOOOCUS and the body parts.
If you can think of anything else, I'd love to hear from you.
Thanks again.

**Reply by Artflowman** [reply]
The penis/testicles-problem can be solved by a picture "transplantation" during a high stadium of the process (a photorealistic one). You have to port a "female male" to a 3rd party program (I use artflow on my phone. Then give your self-made "male male" back into the picture prompt routine. I always have to put 3 identic prompt pics to get a passable result, a pyracanny pic (0,8 + 0,8), 1 imagePrompt pic (0,75 + 0,75) and 1 cpds pic (0,35 + 0,45). The numbers for "StopAt" and "Weight" can vary, so you have to figure out your own sizes. Enjoy :-) and don't forget to visit my artists profile at DA: deviantart.com/rembrandtius.

**Comment by JRprof53** [comment]
Thanks so much for taking the time to offer help.
Would it be possible to explain what you mean by:

 "a picture "transplantation" during a high stadium of the process (a photorealistic one)"    What is "high stadium"?
"port a "female male" to a 3rd party program?  Is this cut and paste?
"port a "female male" to a 3rd party program, Artflow"     Any other apps/software besides Artflow?
"give your self-made "male male" back into the picture prompt routine"    I assume you are referring to the 'image prompt'? And then what steps?*
Any steps where Random or "0" checkbox (seed) is to be used?*

I really appreciate any help you can further offer.
It's been very frustrating.

Thank so much.

**Comment by Artflowman** [comment]
Transplantation:
Means you have to paste a real penis picture at the "whatever they place
there instead" on the SD picture. Choose a good one e. g. from an anatomic
correct picture (medical newspaper, porn picture etc.) BTW, The only model
which can create a dick at all and let him free sight😆 is the latest
juggernaut XL version 10. All other, also filters AND loras, pull 'em pants
over or flatten him to zero.

High stadium:
This means a picture which already is ready and looking fine to selected it
for publishing, with all corrections like deformed or ugly face were
already done. A male dick is now the last missing "dot"at your "i", means
the last act of perfection of your wanted/needed creation. Reason is, that
this body part weren't often/at all modeled during the
training/conditioning process. Therefore there isn't any known useful
keyword for a pictorial representation.

3rd party apps:
Every app or software which is able working in grahic snippets. Ideal is
one that works with layers, photoshop, procreate, GIMP etc.

Give your self made "male male" .. means:
After the graphical steps in your 3rd party app you give this result as
image prompt back to SD.
Please contact me via Deviantart. I will share my first steps then:
www.deviantart.com/therealrembrandtius

JRprof53 ***@***.***> schrieb am So., 5. Mai 2024, 23:28:

> Thanks so much for taking the time to offer help.
> Would it be possible to explain what you mean by:
>
> "a picture "transplantation" during a high stadium of the process (a
> photorealistic one)"port a "female male" to a 3rd party program? *What is
> "high stadium"? Is this cut and paste?*
>
> "port a "female male" to a 3rd party program" *Any other apps/software
> besides Artflow?*
>
> "give your self-made "male male" back into the picture prompt routine"
> *Can
> I assume you are referring to the 'image prompt'? And then what steps?*
>
> *Any steps where Random or "0" checkbox (seed) is to be used?*
>
> I really appreciate any help you can further offer.
> It's been very frustrating.
>
> Thank so much.
>
>

**Comment by JRprof53** [comment]
Thanks so much for taking the time to clarify. I'll try your suggestions.
When and if I'm successful, i hope you don't mind and I'll let you know.

I really appreciate your help and guidance.

**Reply by candlelightzipo88** [reply]
so.... could you do it?

another mortal here trying to decipher how to do this

**Comment by JRprof53** [comment]
So far I have been completely unsuccessful. I'm just hoping someone will
read about this and perhaps fix the issue.

**Reply by JoshPaar** [reply]
search  for:
 Virile XL 
in civitai.com
Its a full Turbo check point, something inusual as most male NSFW are SD 1.5, meaning fully compatible with Fooocus without extras loras etc. I would say the only one.
Beyond that you can use the latest JuggernautX  but is not good enough and you need to support it with loras, but the solution above is simpler

**Reply by steppres** [reply]
It's highly, highly dependent on the checkpoint model. For example, the latest Juggernaut X  XL model is very prone to churning out fully nude images (male or female) regardless of the prompt. There are certain other checkpoint models that have been trained *exclusively* on female imagery. Just how there are models trained on anime or photography. Fighting with the prompt can only do so much; If a model is trained on anime images that's probably what it will churn out. If a model is trained on photographic images, well, you get the picture.

To put it bluntly, the model you're using maybe not even "know" what male genitalia is. A model is only as knowledgeable as whatever was fed into it when it was trained. There are quite a few SD 1.5 and SDXL models that are trained with male subjects exclusively, so you might be better off trying one of them.

Another trick you can play with in Fooocus is refiner. If you find a model that does a good job at creating the subject you want but you really like the final look of a different model, the refiner is useful. You can even use something like an anime model then switch the refiner to a realism model halfway through to give very interesting results that aren't achievable otherwise.

A lot of this is just trial and error; trying things and seeing what happens. But my only advice is if a certain model isn't doing what you want easily, it's time to find a new one. 

Also newer doesn't also mean 'better' when it comes to checkpoint models. There are quite a few older models that are favored among the community because they just work so well and are particularly consistent. Juggernaut XL 6 takes less prompt wrangling to produce what you want versus Juggernaut X XL for example.

Try searching CivitAI for images similar to what you want to create, then check out which model and prompt was used.

(Sorry for the length post) 💁🏻‍♂️

**Comment by JRprof53** [comment]
Okay thank you very much. I'll give it a try and let you know.

**Reply by laroix** [reply]
Hi~ There's a model here can fix the issue. it'll give you a perfect realistic penis.
https://civitai.com/models/80473/pornmaster-pro

**Reply by Salvichito** [reply]
am I too late? can't find anything like that on civitai... where to find it now?

**Reply by AlainHanoul** [reply]
To follow ... i'm also loocking to
